internal/testrunner/runners/system/servicedeployer/terraform.go

```diff
@@ -117,10 +158,28 @@ func (tsd TerraformServiceDeployer) SetUp(inCtxt ServiceContext) (DeployedServic
 	}
 
 	outCtxt.Agent.Host.NamePrefix = "docker-fleet-agent"
+
+	err = addTerraformOutputs(outCtxt)
+	if err != nil {
+		return nil, fmt.Errorf("could not handle terraform output %w", err)
+	}
 	service.ctxt = outCtxt
 	return &service, nil
 }
 
+func createOutputDir(runId string) (string, error) {
+	buildDir, err := builder.BuildDirectory()
+	if err != nil {
+		return "", fmt.Errorf("locating build directory failed %w", err)
+	}
```

Comment:
Instead of using `builder.BuildDirectory()`, it would be better to add a new method into the locationManager specific for this use case.

It should be something like:
https://github.com/elastic/elastic-package/blob/59e4fa3fd20e417fe712b9b08c1c717ac1f6ad05/internal/configuration/locations/locations.go#L89

In that way, that location would be under the stack path.